backend/endpoints/api/controller.js

```diff
@@ -1,40 +1,84 @@
 const _aufstellung = require('../aufstellungen/aufstellung');
-const _element = require('../aufstellungen/element');
 const _roles = require('../../authentication/role');
 const _termin = require('../termine/termin');
+const _db = require('../../db/connector');
+const { reduce } = require('../raids/controller');
 
 module.exports = [
-    {function: setAufstellung, path: '/aufstellung', method: 'post', authed: true},
+    {function: setAufstellung, path: '/aufstellungen', method: 'post', authed: true},
 ];
 
-
 async function setAufstellung(req, authentication) {
-    const invalidData = "{\"error\":\"invalid data\"}";
-    
     const termin = req.body.body.terminId;
     const data = req.body.body.aufstellungen;
 
+    const stmt = 'SELECT count(*) AS count FROM Termin WHERE id = ?';
+    const terminExists = await _db.queryV(stmt, [termin]);
+    if (terminExists[0].count === 0) {
+        return [];
+    }
+
     const role = await _roles.getRoleForTermin(authentication, termin);
-    if (role <= 0) return "{\"error\":\"no permission\"}";
+    if (role <= 0) {
+        return [];
+    }
 
     for (boss of data) {
         let aufstellung = boss.aufstellungId;
         if (boss.aufstellungId == null)
         {
-            if (boss.bossId == null) continue;
-            const allBosses = await _termin.addBoss(termin, boss.bossId)
-            aufstellung = allBosses[allBosses.length - 1].id;
+            if (boss.bossId == null) {
+                continue;
+            }
+            try {
+                let res = await _termin.addBoss(termin, boss.bossId);
+                aufstellung = res.insertId;
+            }
+            catch (e) {
+                //Unable to insert, next boss
+                continue;
+            }
             if (boss.isCM === true || boss.isCM === false) {
-                await _aufstellung.setCM(aufstellung, boss.isCM);
+                try {
+                    await _aufstellung.setCM(aufstellung, boss.isCM);
+                }
+                catch (e) {
+                    //Ignore not possible to set CM.
+                }
             }
         }
+        else
+        {
+            const stmt2 = 'SELECT count(*) AS count FROM Aufstellung WHERE (id = ?) and (fk_termin = ?)';
+            const aufstellungTerminExists = await _db.queryV(stmt2, [aufstellung, termin]);
+            if (aufstellungTerminExists[0].count === 0) {
+                continue;
+            }
+        }
+        if (aufstellung == null) {

```

Comment:
In welchem Fall kann diese Bedingung überhaupt erfüllt sein?

Reply:
nie, ist weg